tests/test_client.py

```diff
@@ -0,0 +1,172 @@
+import pytest
+import testinfra
+import os
+import time
+import subprocess
+from datetime import datetime
+
+
+@pytest.fixture(scope='session')
+def katello_client(request, entities, registration_hostname):
+    image_id = subprocess.check_output(['podman', 'images', '-q', '--filter', f"reference={request.param}"])
+    assert image_id
+
+    container_id = subprocess.check_output(['podman', 'run', '-v', '/dev/log:/dev/log', '--systemd=true', '-dt', request.param, '/sbin/init']).decode().strip()
+    container = testinfra.get_host("podman://" + container_id)
+
+    container.run_expect([0], f"rpm -Uvh http://{registration_hostname}/pub/katello-ca-consumer-latest.noarch.rpm")
+    container.run_expect([0], f"subscription-manager register --force --org={entities['organization_label']} --activationkey=\"{entities['activation_key']}\"")
```

Comment:
While the tests are isolated and run independently now, these two steps here combined take about a minute. That means every test has a runtime of at least a minute. So 10 tests, 3 OSes, that's around half an hour. Previously, when the tests were considered to be order dependent I was getting 6 minute run times. 

On the plus side, I was able to run it with `-n 4` with a run time of 4 minutes.